### PR TITLE
Install verification for composer via script

### PR DESCRIPTION
**Pull request by jlaswell**
Added a simple php script to install composer using a hash provided from
getcomposer.org. Updated the Dockerfiles to use this new script. Also,
created a .dockerignore file to remove the .git directory on builds to
reduce image size.

Closer to real automation!

ezgif-2011680430

### CODE REVIEW

**Comment by bkuhl** [comment]
The best gif I've seen in a PR....

@mabon will be VERY happy

**Comment by bkuhl** [comment]
Can we move `install_composer.php` to `./infrastructure`?

**Comment by jlaswell** [comment]
@bkuhl Sure thing!

**Comment by bkuhl** [comment]
My PR could really use this

**Comment by jlaswell** [comment]
Feel free to update mine!

File: infrastructure/fpm/Dockerfile [hunk 1]
```diff
@@ -8,19 +8,14 @@ ADD . /var/www/html
...
-    && php -r "unlink('composer-setup.php');" \
-
-    # parallel dependency installation
-    && composer global require hirak/prestissimo \
+    && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
+# Install Composer
+RUN php /var/www/html/infrastructure/install_composer.php
+# parallel dependency installation
+RUN composer global require hirak/prestissimo \
```
Comment (bkuhl) [inline infrastructure/fpm/Dockerfile]: Did you intend to leave these as 2 separate steps?

Comment (jlaswell) [inline infrastructure/fpm/Dockerfile]: Yea, there were some issues with letting composer register with the export before running `composer`

**Comment by millidavids** [comment]
👍

**Pull request merged**